thirdparty/assimp/code/Common/StackAllocator.h

```diff
@@ -87,6 +87,13 @@ class StackAllocator {
 
 } // namespace Assimp
 
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
```

Comment:
I had the same problems on MacOS platform. I just replaced g_maxBytesPerBlock with (64 * 1024 * 1024) size in StackAllocator.inl file and it's solved the problem

Reply:
That would work too but would need to be manually added on upgrade. It's clean, but I honestly don't like it as much. It would prevent having to worry about c++17 for sure and can push that. I really don't like putting specific OS's in code like this.

Reply:
Ooof. That's fun.

```
/home/brad/git/thunder/thirdparty/assimp/code/Common/StackAllocator.inl:61:42: error: no matching function for call to 'min'
   61 |         m_blockAllocationSize = std::max(std::min(m_blockAllocationSize * 2, (64 * 1024 * 1024)), byteSize);
```

Reply:
Do you see any other current issues? I just pushed the fix you suggested. Right now, everything else compiles without a hitch and the sdk is linking properly.

```
install-root/sdk/2024.2/freebsd/x86/bin/WorldEditor:
	libengine-editor.so => /home/brad/git/thunder-build/install-root/sdk/2024.2/freebsd/x86/bin/../lib/libengine-editor.so (0x16a8d7400000)
	libQt5Xml.so.5 => /usr/local/lib/qt5/libQt5Xml.so.5 (0x16a8d5c6a000)
	libQt5QuickWidgets.so.5 => /usr/local/lib/qt5/libQt5QuickWidgets.so.5 (0x16a8d468a000)
	libnext-editor.so => /home/brad/git/thunder-build/install-root/sdk/2024.2/freebsd/x86/bin/../lib/libnext-editor.so (0x16a8d4dea000)
	libQt5Widgets.so.5 => /usr/local/lib/qt5/libQt5Widgets.so.5 (0x16a8da600000)
	libQt5Quick.so.5 => /usr/local/lib/qt5/libQt5Quick.so.5 (0x16a8db400000)
	libQt5Gui.so.5 => /usr/local/lib/qt5/libQt5Gui.so.5 (0x16a8d65b9000)
	libQt5QmlModels.so.5 => /usr/local/lib/qt5/libQt5QmlModels.so.5 (0x16a8d9a60000)
	libQt5Qml.so.5 => /usr/local/lib/qt5/libQt5Qml.so.5 (0x16a8d9007000)
	libQt5Network.so.5 => /usr/local/lib/qt5/libQt5Network.so.5 (0x16a8dc452000)
	libQt5Core.so.5 => /usr/local/lib/qt5/libQt5Core.so.5 (0x16a8de800000)
	libc++.so.1 => /lib/libc++.so.1 (0x16a8dd10b000)
	libcxxrt.so.1 => /lib/libcxxrt.so.1 (0x16a8ddaa0000)
	libm.so.5 => /lib/libm.so.5 (0x16a8e06e7000)
	libgcc_s.so.1 => /lib/libgcc_s.so.1 (0x16a8df3f4000)
	libc.so.7 => /lib/libc.so.7 (0x16a8e0bcc000)
	libglfw-editor.so => /home/brad/git/thunder-build/install-root/sdk/2024.2/freebsd/x86/lib/../lib/libglfw-editor.so (0x16a8e007c000)
	libphysfs-editor.so => /home/brad/git/thunder-build/install-root/sdk/2024.2/freebsd/x86/lib/../lib/libphysfs-editor.so (0x16a8e2176000)
	libfreetype-editor.so => /home/brad/git/thunder-build/install-root/sdk/2024.2/freebsd/x86/lib/../lib/libfreetype-editor.so (0x16a8e1277000)
	libQt5Gamepad.so.5 => /usr/local/lib/qt5/libQt5Gamepad.so.5 (0x16a8e3043000)
	libQt5Svg.so.5 => /usr/local/lib/qt5/libQt5Svg.so.5 (0x16a8e3c51000)
	libzlib-editor.so => /home/brad/git/thunder-build/install-root/sdk/2024.2/freebsd/x86/lib/../lib/libzlib-editor.so (0x16a8e451c000)
	libthr.so.3 => /lib/libthr.so.3 (0x16a8e53e6000)
	libGL.so.1 => /usr/local/lib/libGL.so.1 (0x16a8e5c0e000)
	libpng16.so.16 => /usr/local/lib/libpng16.so.16 (0x16a8e6a00000)
	libz.so.6 => /lib/libz.so.6 (0x16a8e8145000)
	libharfbuzz.so.0 => /usr/local/lib/libharfbuzz.so.0 (0x16a8e72b3000)
	libkvm.so.7 => /lib/libkvm.so.7 (0x16a8e962e000)
	libprocstat.so.1 => /usr/lib/libprocstat.so.1 (0x16a8e9139000)
	libexecinfo.so.1 => /usr/lib/libexecinfo.so.1 (0x16a8e9ef6000)
	libdouble-conversion.so.3 => /usr/local/lib/libdouble-conversion.so.3 (0x16a8eb05e000)
	libicui18n.so.74 => /usr/local/lib/libicui18n.so.74 (0x16a8ebeca000)
	libicuuc.so.74 => /usr/local/lib/libicuuc.so.74 (0x16a8ecbf3000)
	libpcre2-16.so.0 => /usr/local/lib/libpcre2-16.so.0 (0x16a8ea197000)
	libzstd.so.1 => /usr/local/lib/libzstd.so.1 (0x16a8ee19e000)
	libglib-2.0.so.0 => /usr/local/lib/libglib-2.0.so.0 (0x16a8ed2e1000)
	libSM.so.6 => /usr/local/lib/libSM.so.6 (0x16a8eee8f000)
	libICE.so.6 => /usr/local/lib/libICE.so.6 (0x16a8efce2000)
	libX11.so.6 => /usr/local/lib/libX11.so.6 (0x16a8f0b31000)
	libXext.so.6 => /usr/local/lib/libXext.so.6 (0x16a8f0c7f000)
	libinotify.so.0 => /usr/local/lib/libinotify.so.0 (0x16a8f1443000)
	libGLdispatch.so.0 => /usr/local/lib/libGLdispatch.so.0 (0x16a8f26ab000)
	libGLX.so.0 => /usr/local/lib/libGLX.so.0 (0x16a8f1eae000)
	libfreetype.so.6 => /usr/local/lib/libfreetype.so.6 (0x16a8f360d000)
	libgraphite2.so.3 => /usr/local/lib/libgraphite2.so.3 (0x16a8f4b6d000)
	libelf.so.2 => /lib/libelf.so.2 (0x16a8f39cf000)
	libutil.so.9 => /lib/libutil.so.9 (0x16a8f4324000)
	libicudata.so.74 => /usr/local/lib/libicudata.so.74 (0x16a8f5686000)
	libiconv.so.2 => /usr/local/lib/libiconv.so.2 (0x16a8f5f66000)
	libintl.so.8 => /usr/local/lib/libintl.so.8 (0x16a8f6b0d000)
	libpcre2-8.so.0 => /usr/local/lib/libpcre2-8.so.0 (0x16a8f8135000)
	libxcb.so.1 => /usr/local/lib/libxcb.so.1 (0x16a8f7554000)
	libdl.so.1 => /usr/lib/libdl.so.1 (0x16a8f83ed000)
	libbz2.so.4 => /usr/lib/libbz2.so.4 (0x16a8f9064000)
	libbrotlidec.so.1 => /usr/local/lib/libbrotlidec.so.1 (0x16a8f987f000)
	libXau.so.6 => /usr/local/lib/libXau.so.6 (0x16a8f9a19000)
	libXdmcp.so.6 => /usr/local/lib/libXdmcp.so.6 (0x16a8fa2c3000)
	libbrotlicommon.so.1 => /usr/local/lib/libbrotlicommon.so.1 (0x16a8faee5000)
	[vdso] (0x16a8d4339000)
```